modules/_labs/select/react/lib/types.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * The placement of the menu relative to its corresponding button. This type is a
+ * subset of the PopperJS.Placement type limited to the `bottom` and `top` values.
+ *
+ * We need to use `top` and `bottom` instead of `top-start` and `bottom-start` for
+ * placements because PopperJS incorrectly rounds the `start` and `end` modifiers:
+ * https://github.com/popperjs/popper-core/blob/38914aae7a2e91715c6eb2b563517082a40cfa64/src/utils/computeOffsets.js#L68-L81
+ * This rounding causes problems with browsers that allow subpixel values for elements
+ * like Firefox and Edge.
+ */
+export type MenuPlacement = 'bottom' | 'top';
+
+/**
+ * The visibility state of the menu.
+ *
+ * `closed`: The menu is completely closed (not present in the DOM).
+ * `open`: The menu was previously `closed` and has just been instructed to open.
+ *   The menu has been added to the DOM, but it's not yet visible. This state is
+ *   necessary in order to apply an opacity of 0 to the menu before transitioning it
+ *   to an opacity of 1.0 in the opening state.
+ * `opening`: The menu is in the process of opening.
+ * `opened`: The menu is completely open.
+ * `close`: The menu was previously `open` and has just been instructed to close.
+ * `closing`: The menu is in the process of closing.
```

Comment:
I'm a bit confused on the difference between `close` and `closing`

Reply:
`close` is basically the opposite of `open`. Whereas `open` indicates the menu is _just now_ transitioning from `closed` and starting to open, `close` indicates the menu is _just now_ transitioning from `opened` and starting to close.

I added `close` mostly to keep things symmetrical, but it's not actually necessary. `open` was necessary to provide a starting point where the menu could be added to the DOM but with an opacity of `0` so that it could be transitioned to an opacity of `1` as it opens. However, we can go straight from `opened` to `closing` without adding the intermediate `close` state since the menu already exists in the DOM with the desired starting opacity of `1` in the `opened` state.